warp10/src/main/java/io/warp10/continuum/store/Directory.java

```diff
@@ -1761,10 +1773,38 @@ public void run() {
             // Clear the cache if it is an update
             //
             
-            if (io.warp10.continuum.Configuration.INGRESS_METADATA_UPDATE_ENDPOINT.equals(metadata.getSource())) {
+            if (io.warp10.continuum.Configuration.INGRESS_METADATA_UPDATE_ENDPOINT.equals(metadata.getSource())
+                || io.warp10.continuum.Configuration.INGRESS_METADATA_UPDATE_DELTA_ENDPOINT.equals(metadata.getSource())) {
               id = MetadataUtils.id(metadata);
               directory.serializedMetadataCache.remove(id);
               
+              //
+              // If this is a delta update of attributes, consolidate them
+              //
+              
+              if (io.warp10.continuum.Configuration.INGRESS_METADATA_UPDATE_DELTA_ENDPOINT.equals(metadata.getSource())) {
+                Metadata meta = directory.metadatas.get(metadata.getName()).get(labelsId);
+                
+                boolean hasChanged = false;
+                
+                for (Entry<String,String> attr: metadata.getAttributes().entrySet()) {
+                  hasChanged = true;
+                  if ("".equals(attr.getValue())) {
+                    meta.getAttributes().remove(attr.getKey());
+                  } else {
+                    meta.putToAttributes(attr.getKey(), attr.getValue());
+                  }
+                }
+                
+                if (hasChanged) {
```

Comment:
Could use `metadata.getAttributesSize() > 0` instead of `hasChanged` and remove that variable

Reply:
The attributes should be updated unconditionally any way...